### ISSUE TITLE
undefined method `locations'

### ISSUE THREAD

**Issue by 22rtz** [cucumber/cucumber-ruby#1386]
## Summary

Running cucumber-ruby 4.0.0rc.3 the following error is raised
```
undefined method `locations' for #<Cucumber::Messages::Pickle::PickleStep:0x00007fb6a98c3ea0> (NoMethodError)
/Users/USERNAME/.rvm/gems/ruby-2.6.3/gems/cucumber-core-5.0.2/lib/cucumber/core/compiler.rb:42:in `method_missing'
/Users/USERNAME/.rvm/gems/ruby-2.6.3/gems/cucumber-core-5.0.2/lib/cucumber/core/compiler.rb:42:in `create_test_step'
/Users/USERNAME/.rvm/gems/ruby-2.6.3/gems/cucumber-core-5.0.2/lib/cucumber/core/compiler.rb:35:in `block in create_test_case'
/Users/USERNAME/.rvm/gems/ruby-2.6.3/gems/cucumber-core-5.0.2/lib/cucumber/core/compiler.rb:35:in `each'
/Users/USERNAME/.rvm/gems/ruby-2.6.3/gems/cucumber-core-5.0.2/lib/cucumber/core/compiler.rb:35:in `collect'
/Users/USERNAME/.rvm/gems/ruby-2.6.3/gems/cucumber-core-5.0.2/lib/cucumber/core/compiler.rb:35:in `create_test_case'
/Users/USERNAME/.rvm/gems/ruby-2.6.3/gems/cucumber-core-5.0.2/lib/cucumber/core/compiler.rb:22:in `pickle'
/Users/USERNAME/.rvm/gems/ruby-2.6.3/gems/cucumber-core-5.0.2/lib/cucumber/core/gherkin/parser.rb:24:in `block in document'
/Users/USERNAME/.rvm/gems/ruby-2.6.3/gems/gherkin-8.2.1/lib/gherkin/stream/parser_message_stream.rb:39:in `yield'
/Users/USERNAME/.rvm/gems/ruby-2.6.3/gems/gherkin-8.2.1/lib/gherkin/stream/parser_message_stream.rb:39:in `block (3 levels) in messages'
/Users/USERNAME/.rvm/gems/ruby-2.6.3/gems/gherkin-8.2.1/lib/gherkin/stream/parser_message_stream.rb:38:in `each'
/Users/USERNAME/.rvm/gems/ruby-2.6.3/gems/gherkin-8.2.1/lib/gherkin/stream/parser_message_stream.rb:38:in `block (2 levels) in messages'
/Users/USERNAME/.rvm/gems/ruby-2.6.3/gems/gherkin-8.2.1/lib/gherkin/stream/parser_message_stream.rb:83:in `yield'
/Users/USERNAME/.rvm/gems/ruby-2.6.3/gems/gherkin-8.2.1/lib/gherkin/stream/parser_message_stream.rb:83:in `block (2 levels) in sources'
/Users/USERNAME/.rvm/gems/ruby-2.6.3/gems/gherkin-8.2.1/lib/gherkin/stream/parser_message_stream.rb:82:in `each'
/Users/USERNAME/.rvm/gems/ruby-2.6.3/gems/gherkin-8.2.1/lib/gherkin/stream/parser_message_stream.rb:82:in `block in sources'
/Users/USERNAME/.rvm/gems/ruby-2.6.3/gems/gherkin-8.2.1/lib/gherkin/stream/parser_message_stream.rb:26:in `each'
/Users/USERNAME/.rvm/gems/ruby-2.6.3/gems/gherkin-8.2.1/lib/gherkin/stream/parser_message_stream.rb:26:in `each'
/Users/USERNAME/.rvm/gems/ruby-2.6.3/gems/gherkin-8.2.1/lib/gherkin/stream/parser_message_stream.rb:26:in `block in messages'
/Users/USERNAME/.rvm/gems/ruby-2.6.3/gems/cucumber-core-5.0.2/lib/cucumber/core/gherkin/parser.rb:20:in `each'
/Users/USERNAME/.rvm/gems/ruby-2.6.3/gems/cucumber-core-5.0.2/lib/cucumber/core/gherkin/parser.rb:20:in `each'
/Users/USERNAME/.rvm/gems/ruby-2.6.3/gems/cucumber-core-5.0.2/lib/cucumber/core/gherkin/parser.rb:20:in `document'
/Users/USERNAME/.rvm/gems/ruby-2.6.3/gems/cucumber-core-5.0.2/lib/cucumber/core.rb:30:in `block in parse'
/Users/USERNAME/.rvm/gems/ruby-2.6.3/gems/cucumber-core-5.0.2/lib/cucumber/core.rb:29:in `each'
/Users/USERNAME/.rvm/gems/ruby-2.6.3/gems/cucumber-core-5.0.2/lib/cucumber/core.rb:29:in `parse'
/Users/USERNAME/.rvm/gems/ruby-2.6.3/gems/cucumber-core-5.0.2/lib/cucumber/core.rb:21:in `compile'
/Users/USERNAME/.rvm/gems/ruby-2.6.3/gems/cucumber-4.0.0.rc.3/lib/cucumber/runtime.rb:73:in `run!'
/Users/USERNAME/.rvm/gems/ruby-2.6.3/gems/cucumber-4.0.0.rc.3/lib/cucumber/cli/main.rb:29:in `execute!'
/Users/USERNAME/.rvm/gems/ruby-2.6.3/gems/cucumber-4.0.0.rc.3/bin/cucumber:9:in `<top (required)>'
/Users/USERNAME/.rvm/gems/ruby-2.6.3/bin/cucumber:23:in `load'
/Users/USERNAME/.rvm/gems/ruby-2.6.3/bin/cucumber:23:in `<main>'
/Users/USERNAME/.rvm/gems/ruby-2.6.3/bin/ruby_executable_hooks:24:in `eval'
/Users/USERNAME/.rvm/gems/ruby-2.6.3/bin/ruby_executable_hooks:24:in `<main>'
```

## Expected Behavior

Cucumber's features are executed correctly.

## Current Behavior

Cucumber's features can't be run.

## Your Environment

* Version used: cucumber 4.0.0.rc.3
* Ruby version: 2.6.3
* Operating System and version: MacOS Mojave 10.14.6

Running the `gem info cucumber`
Output is:
```
*** LOCAL GEMS ***

cucumber (4.0.0.rc.3, 3.1.2)
    Authors: Aslak Hellesøy, Matt Wynne, Steve Tooke
    Homepage: https://cucumber.io/
    License: MIT
    Installed at (4.0.0.rc.3): /Users/USERNAME/.rvm/gems/ruby-2.6.3
                 (3.1.2): /Users/USERNAME/.rvm/gems/ruby-2.6.3

    cucumber-4.0.0.rc.3
```

**Comment by 22rtz** [comment]
```
GEM
  remote: https://rubygems.org/
  specs:
    appium_lib (10.5.0)
      appium_lib_core (~> 3.3)
      nokogiri (~> 1.8, >= 1.8.1)
      tomlrb (~> 1.1)
    appium_lib_core (3.5.0)
      faye-websocket (~> 0.10.0)
      selenium-webdriver (~> 3.14, >= 3.14.1)
    backports (3.15.0)
    builder (3.2.4)
    childprocess (3.0.0)
    cucumber (4.0.0.rc.3)
      builder (~> 3.2, >= 3.2.3)
      cucumber-core (~> 5.0, >= 5.0.0)
      cucumber-expressions (~> 8.0, >= 8.0.2)
      cucumber-wire (~> 1.1, >= 1.1.0)
      diff-lcs (~> 1.3, >= 1.3)
      gherkin (~> 8.1, >= 8.1.1)
      multi_json (~> 1.13, >= 1.13.1)
      multi_test (~> 0.1, >= 0.1.2)
    cucumber-core (5.0.2)
      backports (~> 3.15, >= 3.15.0)
      cucumber-tag_expressions (~> 2.0, >= 2.0.2)
      gherkin (~> 8.1, >= 8.1.1)
    cucumber-expressions (8.3.0)
    cucumber-messages (7.0.0)
      google-protobuf (>= 3.2, <= 3.8)
    cucumber-tag_expressions (2.0.2)
    cucumber-wire (1.1.0)
      cucumber-core (~> 5.0, >= 5.0.0)
      cucumber-expressions (~> 8.0, >= 8.0.2)
    diff-lcs (1.3)
    eventmachine (1.2.7)
    faye-websocket (0.10.9)
      eventmachine (>= 0.12.0)
      websocket-driver (>= 0.5.1)
    ffi (1.12.1)
    ffi-serial (1.0.5)
      ffi (~> 1.9, >= 1.9.3)
    gherkin (8.2.1)
      cucumber-messages (~> 7.0, >= 7.0.0)
    google-protobuf (3.8.0)
    json (2.3.0)
    mini_portile2 (2.4.0)
    multi_json (1.14.1)
    multi_test (0.1.2)
    nokogiri (1.10.7)
      mini_portile2 (~> 2.4.0)
    report_builder (1.8)
      json (>= 1.8.1)
    roo (2.8.2)
      nokogiri (~> 1)
      rubyzip (>= 1.2.1, < 2.0.0)
    rubyzip (1.3.0)
    selenium-webdriver (3.142.7)
      childprocess (>= 0.5, < 4.0)
      rubyzip (>= 1.2.2)
    tomlrb (1.2.9)
    websocket-driver (0.7.1)
      websocket-extensions (>= 0.1.0)
    websocket-extensions (0.1.4)

PLATFORMS
  ruby

DEPENDENCIES
  appium_lib (~> 10.5)
  cucumber (~> 4.0.0.rc.2)
  ffi-serial (~> 1.0.5)
  report_builder (~> 1.8)
  roo (~> 2.8, >= 2.8.2)

BUNDLED WITH
   2.0.2

```

**Comment by 22rtz** [comment]
I think the problem is related to missing the `require 'cucumber'` statement in `env.rb` file.

**Comment by vincent-psarga** [comment]
Hi and thank you for reporting the bug.

This could be fixed by adding this in your Gemfile:

```ruby
gem 'gherkin', '~> 8.1',  '= 8.1.1'
```

The next rc release should come next week and this time I'll make sure we don't get this kind of trouble with a new major release of an underlying library.

**Comment by vincent-psarga** [comment]
This should be fixed by `4.0.0.rc.4`. Please do not hesitate to reopen the issue if it's still happening.

**Comment by 22rtz** [comment]
> Hi and thank you for reporting the bug.
> 
> This could be fixed by adding this in your Gemfile:
> 
> ```ruby
> gem 'gherkin', '~> 8.1',  '= 8.1.1'
> ```
> 
> The next rc release should come next week and this time I'll make sure we don't get this kind of trouble with a new major release of an underlying library.

Sorry for delayed answer. I confirm it works properly with this fix
